app/js/Router.js

```diff
@@ -19,76 +8,94 @@ const SmokeTest = require('smoke-test-sharelatex')
+  app.use(express.static('public'))
+  app.use(SessionMiddleware.middleware)
+  app.use(BodyParser.urlencoded({ extended: false }))
 
-    app.get('/', HttpController.home)
+  app.get('/', HttpController.home)
 
-    app.post(
-      '/login',
-      celebrate({
-        body: Joi.object({
-          email: Joi.string()
-            .trim()
-            .email()
-            .required(),
-          password: Joi.string()
-            .trim()
-            .required()
-        })
-      }),
-      AuthController.login,
-      AuthController.handleLoginErrors
-    )
+  app.post(
+    '/login',
+    celebrate({
+      body: Joi.object({
+        email: Joi.string()
+          .trim()
+          .email()
+          .required(),
+        password: Joi.string()
+          .trim()
+          .required()
+      })
+    }),
+    rateLimit({
+      max: 10,
+      windowMs: 60000,
+      rateLimiterId: 'login'
+    }),
+    AuthController.login,
+    AuthController.handleLoginErrors
+  )
 
-    app.get('/logout', AuthController.logout)
-    app.get('/one-time-login/request', AuthController.oneTimeLoginRequestForm)
+  app.get('/logout', AuthController.logout)
+  app.get('/one-time-login/request', AuthController.oneTimeLoginRequestForm)
 
-    app.post(
-      '/one-time-login/request',
-      celebrate({
-        body: Joi.object({
-          email: Joi.string()
-            .trim()
-            .email()
-            .required()
-        })
-      }),
-      AuthController.oneTimeLoginRequest,
-      AuthController.handleOneTimeLoginRequestErrors
-    )
+  app.post(
+    '/one-time-login/request',
+    celebrate({
+      body: Joi.object({
+        email: Joi.string()
+          .trim()
+          .email()
+          .required()
+      })
+    }),
+    rateLimit({
+      max: 10,
+      windowMs: 60000,
+      rateLimiterId: 'one-time-login-request'
+    }),
+    AuthController.oneTimeLoginRequest,
+    AuthController.handleOneTimeLoginRequestErrors
+  )
 
-    app.get(
-      '/one-time-login',
-      celebrate({
-        query: Joi.object({
-          email: Joi.string()
-            .email()
-            .required(),
-          token: Joi.string()
-            .regex(/^[0-9a-f]{64}$/, 'token')
-            .required()
-        })
-      }),
-      AuthController.oneTimeLogin,
-      AuthController.handleOneTimeLoginErrors
-    )
+  app.get(
+    '/one-time-login',
+    celebrate({
+      query: Joi.object({
+        email: Joi.string()
+          .email()
+          .required(),
+        token: Joi.string()
+          .regex(/^[0-9a-f]{64}$/, 'token')
+          .required()
+      })
+    }),
+    rateLimit({
+      max: 10,
+      windowMs: 60,
+      rateLimiterId: 'one-time-login'
+    }),
+    AuthController.oneTimeLogin,
+    AuthController.handleOneTimeLoginErrors
+  )
 
-    app.get('/project', HttpController.projects)
-    app.get('/project/:project_id', HttpController.getProject)
+  app.get('/project', HttpController.projects)
```

Comment:
This isn't a rate limiting issue but it seems clearer to use some authentication middleware on the project `GET` endpoints, rather than having the multiple checks for `req.session.user_id`  in each method in HttpController ([HttpController.js#L30-L39](https://github.com/overleaf/read-only/blob/dce5b1bfcf22d874de89156e12ff013720be9370/app/js/HttpController.js#L30-L39) and [HttpController.js#L62-L73](https://github.com/overleaf/read-only/blob/dce5b1bfcf22d874de89156e12ff013720be9370/app/js/HttpController.js#L62-L73)). e.g. `AuthController.requireLogin` 
```
app.get('/project', AuthController.requireLogin, HttpController.projects)
app.get('/project/:project_id', AuthController.requireLogin, 
    AuthController.ensureUserOwnsProject, HttpController.getProject)
```

Equivalent code in web [AuthenticationController.js#L295-L309](https://github.com/overleaf/web-internal/blob/163080b499a7db66852cc92251c2999f49c3ab83/app/src/Features/Authentication/AuthenticationController.js#L295-L309)  and  [AuthorizationMiddleware.js](https://github.com/overleaf/web-internal/blob/ed8636630c1debcb21cea16e50adddcb636631af/app/src/Features/Authorization/AuthorizationMiddleware.js).

Reply:
I'll take note of this and do it in a separate PR.